src/analyzer/waverubber.cpp

```diff
@@ -0,0 +1,95 @@
+/*
+   Strawberry Music Player
+   This file was part of Clementine.
+   Copyright 2004, Melchior FRANZ <mfranz@kde.org>
+   Copyright 2009-2010, David Sansome <davidsansome@gmail.com>
+   Copyright 2010, 2014, John Maguire <john.maguire@gmail.com>
+   Copyright 2014-2015, Mark Furneaux <mark@furneaux.ca>
+   Copyright 2014, Krzysztof Sobiecki <sobkas@gmail.com>
+   Copyright 2024, Gustavo L Conte <suporte@gu.pro.br>
+
+   Strawberry is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   Strawberry is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with Strawberry.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <QPainter>
+#include <QResizeEvent>
+#include "engine/enginebase.h"
+#include "waverubber.h"
+
+const char *WaveRubber::kName = QT_TRANSLATE_NOOP("AnalyzerContainer", "WaveRubber");
+
+WaveRubber::WaveRubber(QWidget *parent)
+  : AnalyzerBase(parent, 9) {}
+
+void WaveRubber::resizeEvent(QResizeEvent *e) {
+  Q_UNUSED(e)
+  canvas_ = QPixmap(size());
+  canvas_.fill(palette().color(QPalette::AlternateBase));
+}
+
+void WaveRubber::analyze(QPainter &p, const Scope &s, bool new_frame) {
+  if (!new_frame || engine_->state() == EngineBase::State::Paused) {
+    p.drawPixmap(0, 0, canvas_);
+    return;
+  }
+
+  // Clear the canvas
+  canvas_ = QPixmap(size());
+  canvas_.fill(palette().color(QPalette::Window));
+
+  QPainter canvas_painter(&canvas_);
+
+  // Set the pen color to the QT palette highlight color
+  canvas_painter.setPen(palette().color(QPalette::Highlight));
+  // Get pointer to amplitude data 
+  const float* amplitude_data = s.data();
+  
+  
+  int mid_y = height() / 4;
+  int num_samples = s.size();
+  
+  float x_scale = static_cast<float>(width()) / num_samples;
+  float prev_y = mid_y;
+
+  // Draw the waveform
+  for (int i = 0; i < num_samples; ++i) {
+    	// Normalize amplitude to 0-1 range
```

Comment:
Indent should be 2 spaces and no tabs, you can use clang-format to automatically format this, but don't run it on all sources as some things with clang-format formatting is wrong.:

```
sed -i 's/DisableFormat:   true/DisableFormat:   false/g' .clang-format 
clang-format -i src/analyzer/waverubber.*
```